### ISSUE TITLE
Implement bar aggregation in Rust

### ISSUE THREAD

**Issue by cjdsellers** [nautechsystems/nautilus_trader#1855]
Implement `BarAggregator`s and add to `DataEngine`.

**Comment by cjdsellers** [comment]
A decent base has been written.

Handing over to @faysou for now 🙏.

**Comment by faysou** [comment]
Yes, I'm working on this now, I have some bugs in tests I need to solve, should be ready some time this week.

**Comment by cjdsellers** [comment]
Closing this pending some further movement from @stastnypremysl 

We can open more specific tickets when/if necessary.

Some aggregation methods still remain to be implemented per https://github.com/nautechsystems/nautilus_trader/issues/2603.
